frontend/src/components/common/Terminal.tsx

```diff
@@ -85,24 +123,88 @@ export default function Terminal(props: TerminalProps) {
     }
 
     const encoded = encoder.encode(data);
-    const buffer = new Uint8Array([0, ...encoded]);
+    const buffer = new Uint8Array([channel, ...encoded]);
 
     socket.send(buffer);
   }
 
+  // Channels:
+  // 0: stdin
+  // 1: stdout
+  // 2: stderr
+  // 3: server error
+  // 4: resize channel
   function onData(xterm: XTerminal, bytes: ArrayBuffer) {
+    // Only show data from stdout, stderr and server error channel.
+    const channel = new Int8Array(bytes.slice(0, 1))[0];
+    if (channel !== 1 && channel !== 2 && channel !== 3) {
+      return;
+    }
+
     // The first byte is discarded because it just identifies whether
     // this data is from stderr, stdout, or stdin.
     const data = bytes.slice(1);
     const text = decoder.decode(data);
 
-    if (bytes.byteLength < 2) {
+    if (automaticallySwitchShell(xterm, channel, text)) {
       return;
     }
 
+    // send resize command to server once connection is establised
+    if (!(xterm as any).connected && !!text) {
+      xterm.clear();
+      (async function () {
+        send(4, `{"Width":${xterm.cols},"Height":${xterm.rows}}`);
+      })();
+      (xterm as any).connected = true;
+      console.log('Connected!!!');
+    }
+
     xterm.write(text);
   }
 
+  function isShellNotFoundError(channel: number, text: string): boolean {
+    // Linux container Error
+    if (channel === 3) {
+      try {
+        const error = JSON.parse(text);
+        if (error.code === 500 && error.status === 'Failure' && error.reason === 'InternalError') {
+          if (error.details.causes[0].message.includes('executable file not found')) {
+            return true;
+          }
+        }
+      } catch {}
+    }
+    // Windows container Error
+    if (channel === 1) {
+      if (text.includes('The system cannot find the file specified')) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  function automaticallySwitchShell(xterm: XTerminal, channel: number, text: string): boolean {
+    if ((xterm as any).connected) {
+      return false;
+    }
+    if (!shells.automaticallySwitch) {
+      return false;
+    }
+
+    if (isShellNotFoundError(channel, text)) {
+      if (shells.currentIdx + 1 >= shells.available.length) {
+        xterm.write('Could not estable connection.');
+      } else {
+        xterm.write('Retrying connectting with shell: ' + shells.available[shells.currentIdx]);
```

Comment:
There's a typo at `connectting` -> `connecting`.
And this message should be translatable.